library/pk_wrap.c

```diff
@@ -127,6 +127,81 @@ static int rsa_sign_wrap( void *ctx, mbedtls_md_type_t md_alg,
                                     hash, sig ) );
 }
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+static int rsa_decrypt_wrap( void *ctx,
+                    const unsigned char *input, size_t ilen,
+                    unsigned char *output, size_t *olen, size_t osize,
+                    int (*f_rng)(void *, unsigned char *, size_t), void *p_rng )
+{
+    mbedtls_rsa_context * rsa = (mbedtls_rsa_context *) ctx;
+    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+    mbedtls_svc_key_id_t key_id = MBEDTLS_SVC_KEY_ID_INIT;
+    psa_status_t status;
+    mbedtls_pk_context key;
+    int key_len;
+    /* see RSA_PRV_DER_MAX_BYTES in pkwrite.c */
```

Comment:
Same as for encryption. Please consider moving `RSA_PRV_DER_MAX_BYTES` to header file.

Reply:
```
Naming patterns of macros: FAIL
    > include/mbedtls/pk.h:796: 'RSA_PUB_DER_MAX_BYTES' does not match the
      required pattern '^(MBEDTLS|PSA)_[0-9A-Z_]*[0-9A-Z]$'.
      |
  796 | #define RSA_PUB_DER_MAX_BYTES   ( 38 + 2 * MBEDTLS_MPI_MAX_SIZE )
      |         ^^^^^^^^^^^^^^^^^^^^^
    > include/mbedtls/pk.h:813: 'MPI_MAX_SIZE_2' does not match the required
      pattern '^(MBEDTLS|PSA)_[0-9A-Z_]*[0-9A-Z]$'.
      |
  813 | #define MPI_MAX_SIZE_2          ( MBEDTLS_MPI_MAX_SIZE / 2 + \
      |         ^^^^^^^^^^^^^^
    > include/mbedtls/pk.h:815: 'RSA_PRV_DER_MAX_BYTES' does not match the
      required pattern '^(MBEDTLS|PSA)_[0-9A-Z_]*[0-9A-Z]$'.
      |
  815 | #define RSA_PRV_DER_MAX_BYTES   ( 47 + 3 * MBEDTLS_MPI_MAX_SIZE \
      |         ^^^^^^^^^^^^^^^^^^^^^
    > include/mbedtls/pk.h:820: 'RSA_PUB_DER_MAX_BYTES' does not match the
      required pattern '^(MBEDTLS|PSA)_[0-9A-Z_]*[0-9A-Z]$'.
      |
  820 | #define RSA_PUB_DER_MAX_BYTES   0
      |         ^^^^^^^^^^^^^^^^^^^^^
    > include/mbedtls/pk.h:821: 'RSA_PRV_DER_MAX_BYTES' does not match the
      required pattern '^(MBEDTLS|PSA)_[0-9A-Z_]*[0-9A-Z]$'.
      |
  821 | #define RSA_PRV_DER_MAX_BYTES   0
```
Should I rename these ?

Reply:
I missed this comment and left similar below.
I think yes. The changes should impact only one source file.

Reply:
Yes, sorry I forgot to mention because it's so deeply ingrained: all public symbols must be prefixed with `MBEDTLS_` or `mbedtls_` (depending on the type of symbol). Usually this is followed by the name of the "module" that the symbol belongs to  (typically the name of the header file it's declared it); this is not always true for old things, but we should try to follow that rule for new things). For private symbols (static functions, etc) we can omit the `MBEDTLS_` prefix.

So, ideally this should be renamed to start with `MBEDTLS_PK_` since it's declared in `pk.h`.

Reply:
Will be done in #5581 instead of currently pushed solution, will need to rebase when merged